client/gui-qt/icons.cpp

```diff
@@ -39,21 +44,84 @@ fcIcons *fcIcons::instance()
  */
 void fcIcons::drop() { NFCN_FREE(m_instance); }
 
+namespace {
+
+class icon_engine : public QIconEngine {
+public:
+  explicit icon_engine(const QString &path) : path(path) {}
+
+  icon_engine *clone() const override { return new icon_engine(path); }
+
+  void paint(QPainter *painter, const QRect &rect, QIcon::Mode mode,
+             QIcon::State state) override
```

Comment:
I guess this is only ever used with `state = QIcon::On`? Do we want a different color for `mode = QIcon::Active`, maybe the active text color (`QPalette::Active`)?
Add `Q_UNUSED` to avoid compiler warnings.